sdks/java/core/src/test/java/org/apache/beam/sdk/io/AvroIOTest.java

```diff
@@ -157,6 +160,46 @@ public void testAvroIOWriteAndReadASingleFile() throws Throwable {
     p.run();
   }
 
+  /** Options with ValueProviders for testing runtime support. */
+  public interface RuntimeTestOptions extends PipelineOptions {
+    ValueProvider<String> getOutput();
+    void setOutput(ValueProvider<String> output);
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testAvroIOWriteAndReadASingleRuntimeFile() throws Throwable {
+    List<GenericClass> values = ImmutableList.of(new GenericClass(3, "hi"),
+        new GenericClass(5, "bar"));
+    File outputFile = tmpFolder.newFile("output.avro");
+    final String outputPath = outputFile.getAbsolutePath();
+
+    RuntimeTestOptions options = p.getOptions().as(RuntimeTestOptions.class);
+    p.apply(Create.of(values))
+        .apply(AvroIO.write(GenericClass.class)
+            .to(new ValueProvider<String>() {
+              @Override
+              public String get() {
+                return outputPath;
+              }
+
+              @Override
+              public boolean isAccessible() {
+                return false;
+              }
+            })
+            .withoutSharding());
+    p.run();
+
+    PCollection<GenericClass> input =
```

Comment:
reusing the same pipeline like this can cause errors/unexpected behavior. Unless there's something specific about the unit test environment that makes it okay, I wouldn't do it.